### ISSUE TITLE
Accurate Metadataset Epoch

### ISSUE THREAD

**Issue by lvoegtle** [NVIDIA/Megatron-Energon#14]
For LLMs, it is a common methodology, to show every sample exactly once (or n times). The metadataset currently only implements blending/mixing with probability. As a workaround, the sampling probability could be set to the size of the respective dataset, but that doesn't ensure that every sample is seen exactly once (considering shuffle buffer, and repeating) and it doesn't stop the dataset iteration.

A suggestion for an additional interface:

```py
# epochized=True makes the dataset not repeat, but uses the underlying metadataset definition to stop iterating. Needs handling for non-complete batches
def get_train_dataset(..., repeat: bool = True):
```

And in the dataset:
```yaml
__module__: megatron.energon
__class__: MetadatasetV2
splits:
  train:
    # If blend_epochized is used, do not weight the dataset, but use repetitions for, i.e. the weight is effectively repetitions multiplied with the dataset size
    blend_epochized:
      - repetitions: 1
        path: ...
      - ...
```

If `weight_samples=True` (also for nested datasets), and the dataset weights are all integers, then `epochized=True` may be set for `get_train_dataset`, implicating the following constraint: `get_train_dataset` stops iterating after every sample in the datasets has been seen respective `weight` times.

Implementation thoughts:
* Set `repeat=False` for the `WebdatasetSampleLoaderDataset`. Maybe this parameter can be removed entirely. Makes the inner dataset stop after one epoch.
* Implement new `BlendDataset` wrapper, which ends after one epoch and does not loop inner datasets
